replay/models/hnswlib.py

```diff
@@ -192,6 +194,7 @@ def build_index(iterator: Iterator[pd.DataFrame]):
 
                 # saving index to local temp file and sending it to executors
                 temp_path = tempfile.mkdtemp()
+                weakref.finalize(self, shutil.rmtree, temp_path)
```

Comment:
Now it is ok. But we need to research more this question in relation to standard Spark mechanism.
Take a look at Synapse ML lightgbm. There should be an example of deleting directories at the end of Spark app.